test/debugger.js

```diff
@@ -0,0 +1,266 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/*!
+ * @module debug/debugger
+ */
+
+var pjson = require('../package.json');
+var assert = require('assert');
+var common = require('@google-cloud/common');
+var qs = require('querystring');
+var util = require('util');
+var Debug = require('../src');
+
+/** @const {string} Cloud Debug API endpoint */
+var API = 'https://clouddebugger.googleapis.com/v2/debugger';
+
+/**
+ * @constructor
+ */
+function Debugger() {
+  common.ServiceObject.call(this, {
+    parent: new Debug({}),
+    baseUrl: '/debugger'
+  });
+
+  /** @private {string} */
+  this.nextWaitToken_ = null;
+
+  this.clientVersion_ = pjson.name + '/default/v' + pjson.version;
+}
+
+util.inherits(Debugger, common.ServiceObject);
+
+/**
+ * Gets a list of debuggees in a given project to which the user can set
+ * breakpoints.
+ * @param {string} projectId - The project ID for the project whose debuggees
+ *     should be listed.
+ * @param {boolean=} includeInactive - Whether or not to include inactive
+ *     debuggees in the list (default false).
+ * @param {!function(?Error,Debuggee[]=)} callback - A function that will be
+ *     called with a list of Debuggee objects as a parameter, or an Error object
+ *     if an error occurred in obtaining it.
+ */
+Debugger.prototype.listDebuggees = function(projectId, includeInactive, callback) {
+  if (typeof(includeInactive) === 'function') {
+    callback = includeInactive;
+    includeInactive = false;
+  }
+
+  var query = {
+    clientVersion: this.clientVersion_,
+    includeInactive: includeInactive,
+    project: projectId
+  };
+
+  var uri = API + '/debuggees?' + qs.stringify(query);
+  this.request({uri: uri, json: true}, function(err, body, response) {
+    if (err) {
+      callback(err);
+    } else if (!response) {
+      callback(new Error('unknown error - request response missing'));
+    } else if (response.statusCode !== 200) {
+      callback(new Error('unable to list debuggees, status code ' +
+                         response.statusCode));
+    } else if (!body) {
+      callback(new Error('invalid response body from server'));
+    } else {
+      if (body.debuggees) {
+        callback(null, body.debuggees);
+      } else {
+        callback(null, []);
+      }
+    }
+  });
+};
+
+/**
+ * Gets a list of breakpoints in a given debuggee.
+ * @param {string} debuggeeId - The ID of the debuggee whose breakpoints should
```

Comment:
Personally I think it'd be better if we accepted both a debuggee object or simply its ID, and read the `id` property if it turns out to be an object (same goes for breakpoints). I went with just IDs first because `getBreakpoint` should take a breakpoint ID, and everything else followed suit. What do you think about making it accept either?